exporters/trace_stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverV1ExporterHandler.java

```diff
@@ -132,6 +155,67 @@ private static String encodeTraceId(TraceId traceId) {
     return TraceSpan.SpanKind.SPAN_KIND_UNSPECIFIED;
   }
 
+  private static String renderNetworkEvents(TimedEvent<NetworkEvent> timedEvent, Timestamp start) {
+    StringBuilder stringBuilder = new StringBuilder();
+    renderDelay(stringBuilder, timedEvent.getTimestamp().subtractTimestamp(start));
+    NetworkEvent networkEvent = timedEvent.getEvent();
+    if (networkEvent.getType() == Type.RECV) {
+      stringBuilder.append("Received");
+    } else if (networkEvent.getType() == Type.SENT) {
+      stringBuilder.append("Sent");
+    } else {
+      stringBuilder.append("Unknown");
+    }
+    stringBuilder.append(" message_id=");
+    stringBuilder.append(networkEvent.getMessageId());
+    stringBuilder.append(" uncompressed_size=");
+    stringBuilder.append(networkEvent.getUncompressedMessageSize());
+    stringBuilder.append(" compressed_size=");
+    stringBuilder.append(networkEvent.getCompressedMessageSize());
+    if (networkEvent.getKernelTimestamp() != null) {
+      stringBuilder.append(" kernel_timestamp=");
+      stringBuilder.append(networkEvent.getKernelTimestamp().toString());
+    }
+    return stringBuilder.toString();
+  }
+
+  private static String renderAnnotation(TimedEvent<Annotation> timedEvent, Timestamp start) {
+    StringBuilder stringBuilder = new StringBuilder();
+    renderDelay(stringBuilder, timedEvent.getTimestamp().subtractTimestamp(start));
+    Annotation annotation = timedEvent.getEvent();
+    stringBuilder.append(annotation.getDescription());
+    if (!annotation.getAttributes().isEmpty()) {
+      stringBuilder.append(" ");
+      stringBuilder.append(renderAttributes(annotation.getAttributes()));
+    }
+    return stringBuilder.toString();
+  }
+
+  private static void renderDelay(StringBuilder stringBuilder, Duration delay) {
+    long delayUs = delay.getSeconds() * 1000000 + delay.getNanos() / 1000;
+    stringBuilder.append("[@");
+    stringBuilder.append(delayUs);
+    stringBuilder.append(" us] ");
+  }
+
+  private static String renderAttributes(Map<String, AttributeValue> attributes) {
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append("Attributes:{");
+    boolean first = true;
+    for (Map.Entry<String, AttributeValue> entry : attributes.entrySet()) {
+      if (first) {
+        first = false;
+      } else {
+        stringBuilder.append(", ");
+      }
+      stringBuilder.append(entry.getKey());
+      stringBuilder.append("=");
+      stringBuilder.append(attributeValueToString(entry.getValue()));
```

Comment:
Do these values have a chance of needing json escaping?

Reply:
Using proto + gRPC so no need to json escape them 👍